target/linux/ramips/dts/DIR-510L.dts

```diff
@@ -0,0 +1,160 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7620a.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	compatible = "dlink,dir-510l", "ralink,mt7620a-soc";
+	model = "D-Link DIR-510L";
+
+	aliases {
+		led-boot = &led_status;
+		led-failsafe = &led_status;
+		led-running = &led_status;
+		led-upgrade = &led_status;
+	};
+
+	chosen {
+		bootargs = "console=ttyS1,57600";
+	};
+
+	gpio-keys-polled {
+		compatible = "gpio-keys-polled";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		poll-interval = <20>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 1 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 2 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		led_status: status {
+			label = "dir-510l:green:status";
+			gpios = <&gpio0 14 GPIO_ACTIVE_HIGH>;
+		};
+
+		status-red {
+			label = "dir-510l:red:status";
+			gpios = <&gpio0 11 GPIO_ACTIVE_HIGH>;
+		};
+
+	};
+
+	gpio_export {
```

Comment:
@CHKDSK88 Why is this gpio_export needed exactly?

Would it be possible to replace the gpio-export construct with the userspace gpio-switch equivalent (If you need a example device, please look at: fa3c267) ? Or ff you need to toggle some GPIO during startup, you should use gpio hog for this.

Reply:
@ynezz 

It is used for USB ports enable.

I check recent supported devices and gpio-export is still used in ramips:
https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/dts/TUBE-E4G.dts
https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/dts/CreativeBox-v1.dts
https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/dts/DWR-922-E2.dts

Reply:
https://github.com/openwrt/openwrt/pull/1366#issuecomment-424834448

Reply:
@ynezz 

Compiled and tested. I will push changes at Friday morning.